great_expectations/datasource/fluent/sources.py

```diff
@@ -597,21 +584,9 @@ def add_or_update_datasource(
         )
         return add_or_update_datasource
 
-    def create_delete_crud_method(
-        self,
-        datasource_type: Type[Datasource],
-        doc_string: str = "",
-    ) -> Callable[[str], None]:
-        def delete_datasource(name: str) -> None:
-            logger.debug(f"Delete {datasource_type} with {name}")
-            self._validate_current_datasource_type(name, datasource_type)
-            self._data_context._delete_fluent_datasource(datasource_name=name)
-            self._data_context._save_project_config()
-
-        delete_datasource.__doc__ = doc_string
-        # attr-defined issue https://github.com/python/mypy/issues/12472
-        delete_datasource.__signature__ = inspect.signature(delete_datasource)  # type: ignore[attr-defined]
-        return delete_datasource
+    def delete_datasource(self, name: str) -> None:
+        self._data_context._delete_fluent_datasource(datasource_name=name)
+        self._data_context._save_project_config()
```

Comment:
Should this just call `self._data_context.delete_datasource()`?

Reply:
This currently does not fail if the target datasource does not exist

Reply:
> Should this just call self._data_context.delete_datasource()?
I think so but I can't remember why we didn't just do that in the first place. 🤔

Reply:
Also you'll need to add this to `great_expectations/datasource/fluent/sources.pyi`

Reply:
It also might be worth discussing if we should have this method at all.
Should we just always expect users to use `context.delete_datasource()`? rather than have ` context.sources.delete_datasource` and `context.delete_datasource()`???

Still.... 2 delete methods is an improvement over 10+ 😅